app/controllers/homes_controller.rb

```diff
@@ -1,4 +1,22 @@
 class HomesController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :authorized]
+  before_action :authorize_user, except: [:index, :authenticated]
```

Comment:
same idea here - both methods check if a user is signed in, so we don't need to repeat actions between the 2. `authenticate_user!` just checks if a user is signed in
`authorize_user` _also_ checks if they are an admin

for `index`, looks like you want just `authenticate_user!` to check if they're signed in, not caring if they are an admin